README.md

```diff
@@ -1,7 +1,130 @@
 
-This module provides the following Vega data transform:
+[![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=popout)](https://github.com/prettier/prettier)
 
-- **Label**
\ No newline at end of file
+## Syntax for label transform
+
+```json
+"transform": [
+  {
+    "type": "label",
+    "size": [800, 500],
+    "padding": 0,
+    "anchor": [
+      "top-left",
+      "left",
+      "bottom-left",
+      "top",
+      "bottom",
+      "top-right",
+      "right",
+      "bottom-right",
+    ],
+    "offset": [ 1, 1, 1, 1, 1, 1, 1, 1 ],
+    "sort": { "field": "datum.year" },
+    "avoidMarks": [ "basePoint", "baseLine" ],
+    "avoidBaseMark": true,
+    "lineAnchor": "end",
+    "markIdx": 0,
+  }
+]
+```
+
+- `sort`: order of label to be placed (**greater** will be placed **after**)
+
+- `size`: size of the chart in format `[width, height]`. **This size have to match with the chart size**
+
+- `anchor` and `offset`: parallel arrays of anchor points and offset values.
+  - `anchor`: list of anchor points of labels to its mark's bounding box
+    - From the example above, for each label, Vega-Label will try to place it at the `middle` first, relative to its mark.
+    - If it collide with some other mark or label, Vega-Label will try to place it at the `right`, relative to its mark.
+
+  - `offset`: list of offset values from the bounding box of the **base mark**.
+    - From the example above, Vega-Label will try to place label with no offset.
+      - **Note**: if the parallel anchor is `middle`, the offset value should be `null`. Otherwise, the value is ignored.
+    - If it cannot place the label, Vega-Label will try to place label with offset value 2.
+    - If it cannot place the label, Vega-Label will try to place label with offset value 1 inside its mark.
+      - **Note**: label will be placed inside its mark if offset is negative.
+
+- `avoidMarks`: list of data of mark; labels will not collide with these marks
+
+- `padding`: the amount of pixels label can extend pass chart bounding box.
+
+- `avoidBaseMark`: a flag specifying if labels are avoiding base mark (from reactive geometry) or not.
+  - If this flag is `false`, Vega-Label only uses base mark to calculate anchor points for label but not mark to be avoided.
+
+- `lineAnchor`: an anchor point for group line mark can be `begin` or `end`
+  - If `lineAnchor` is `begin`, label is at the beginning of the line. Otherwise, label is at the end of the line.
+  - **Note**: this flag only work with group line mark as a base mark using reactive geometry.
+
+- `markIdx`: use when the reactive geometry's mark type is `'group'`
+  - To specify which mark in the group is to used to calculate anchor points for labels.
+
+- label transform has to be used with reactive geometry to use it as base mark to calculate positions of label
+  - Right now, reactive geometry works with `symbol`, `line`, `rect`, and `group` of `line` and `area`
```

Comment:
Does "reactive geometry" really only work with these marks?  
Or "this transform" only works with these marks? 

Please also file another issue discussing what are the marks that are not supported and what are challenges that block them? (I think it's fine to not implement some of them, but it would be great to understand what's the roadblock.)